glib/src/object.rs

```diff
@@ -1524,72 +1604,124 @@ pub trait ObjectExt: ObjectType {
     /// were provided.
     ///
     /// If the signal has a return value then this is returned here.
     #[doc(alias = "g_signal_emitv")]
-    fn emit(&self, signal_id: SignalId, args: &[&dyn ToValue]) -> Result<Option<Value>, BoolError>;
+    fn try_emit(
+        &self,
+        signal_id: SignalId,
+        args: &[&dyn ToValue],
+    ) -> Result<Option<Value>, BoolError>;
+
+    /// Similar to `try_emit` but panics instead of failing.
+    fn emit(&self, signal_id: SignalId, args: &[&dyn ToValue]) -> Option<Value>;
+
     /// Same as `emit` but takes `Value` for the arguments.
-    fn emit_with_values(
+    fn try_emit_with_values(
         &self,
         signal_id: SignalId,
         args: &[Value],
     ) -> Result<Option<Value>, BoolError>;
 
+    /// Similar to `try_emit_with_values` but panics instead of failing.
+    fn emit_with_values(&self, signal_id: SignalId, args: &[Value]) -> Option<Value>;
+
     /// Emit signal by its name.
     ///
     /// This fails if the signal does not exist, the wrong number of arguments is provided, or
     /// arguments of the wrong types were provided.
     ///
     /// If the signal has a return value then this is returned here.
     #[doc(alias = "g_signal_emit_by_name")]
-    fn emit_by_name<'a, N: Into<&'a str>>(
+    fn try_emit_by_name<'a, N: Into<&'a str>>(
         &self,
         signal_name: N,
         args: &[&dyn ToValue],
     ) -> Result<Option<Value>, BoolError>;
 
+    /// Similar to `try_emit_by_name` but panics instead of failing.
+    fn emit_by_name<'a, N: Into<&'a str>>(
+        &self,
+        signal_name: N,
+        args: &[&dyn ToValue],
+    ) -> Option<Value>;
+
     /// Emit signal by its name.
     ///
     /// This fails if the signal does not exist, the wrong number of arguments is provided, or
     /// arguments of the wrong types were provided.
     ///
     /// If the signal has a return value then this is returned here.
-    fn emit_by_name_with_values<'a, N: Into<&'a str>>(
+    fn try_emit_by_name_with_values<'a, N: Into<&'a str>>(
         &self,
         signal_name: N,
         args: &[Value],
     ) -> Result<Option<Value>, BoolError>;
 
+    /// Similar to `try_emit_by_name_with_values` but panics instead of failing.
+    fn emit_by_name_with_values<'a, N: Into<&'a str>>(
+        &self,
+        signal_name: N,
+        args: &[Value],
+    ) -> Option<Value>;
+
     /// Emit signal by signal id with details.
     ///
     /// This fails if the wrong number of arguments is provided, or arguments of the wrong types
     /// were provided.
     ///
     /// If the signal has a return value then this is returned here.
-    fn emit_with_details(
+    fn try_emit_with_details(
         &self,
         signal_id: SignalId,
         details: Quark,
         args: &[&dyn ToValue],
     ) -> Result<Option<Value>, BoolError>;
 
+    /// Similar to `try_emit_with_details` but panics instead of failing.
```

Comment:
Maybe it would be good to have the long docs on the function we expect people to use

Reply:
And make the other ones properly link to the documented function. Needs to be surrounded by `[]` I guess

Reply:
yeah i will look into improving the docs slightly a bit once we get green CI

Reply:
Should be good enough now